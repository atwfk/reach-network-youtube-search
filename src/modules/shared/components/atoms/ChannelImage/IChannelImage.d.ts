declare namespace IChannelImage {
  export interface IProps {
    src: string;
    alt: string;
    width: string;
    height: string;
  }
}

export { IChannelImage };
