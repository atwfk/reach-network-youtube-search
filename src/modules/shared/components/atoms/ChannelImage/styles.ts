import styled from "styled-components";

export const StyledImageContainer = styled.div`
  .image-layer {
    display: flex;
    justify-content: center;
    align-items: center;

    img {
      border-radius: 50%;
    }
  }
`;
