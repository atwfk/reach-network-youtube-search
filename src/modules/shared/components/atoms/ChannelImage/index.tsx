import React from "react";
import type { FC, ReactElement } from "react";
import { IChannelImage } from "./IChannelImage";
import { StyledImageContainer } from "./styles";

const ChannelImage: FC<IChannelImage.IProps> = ({ src, alt, width, height }): ReactElement => {
  return (
    <StyledImageContainer className="image-container">
      <div className="image-layer" style={{ width: `${width}px` }}>
        <img src={src} alt={alt} width={height} height={height} data-testid="channel-image" />
      </div>
    </StyledImageContainer>
  );
};

export default ChannelImage;
