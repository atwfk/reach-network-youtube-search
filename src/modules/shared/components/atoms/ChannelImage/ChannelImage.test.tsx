import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import ChannelImage from ".";

afterEach(cleanup);

describe("ChannelImage Component snapshots", () => {
  it("Should render simple ChannelImage component", () => {
    const tree = renderer
      .create(
        <ChannelImage
          src="https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"
          alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
          width="320"
          height="180"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
