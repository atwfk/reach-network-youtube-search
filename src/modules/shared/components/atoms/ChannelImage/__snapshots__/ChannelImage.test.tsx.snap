// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelImage Component snapshots Should render simple ChannelImage component 1`] = `
.c0 .image-layer {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .image-layer img {
  border-radius: 50%;
}

<div
  className="c0 image-container"
>
  <div
    className="image-layer"
    style={
      Object {
        "width": "320px",
      }
    }
  >
    <img
      alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
      data-testid="channel-image"
      height="180"
      src="https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"
      width="180"
    />
  </div>
</div>
`;
